pkg/pillar/Makefile

```diff
@@ -76,6 +76,12 @@ build-docker-$(APPS): $(DISTDIR)
 build-docker:
 	docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) .
 
+build-docker-dev:
+	docker build $(DOCKER_ARGS) -t $(DOCKER_TAG) . --target build
+
+enter-docker-dev: build-docker-dev
+	docker run -it --rm --entrypoint /bin/sh $(DOCKER_TAG)
```

Comment:
I normally add:
```
 -v $(pwd):$(pwd) 
```
this way I don't have to rebuild the container if I changed a file

Reply:
And how do you use it then? As far as I understand it will mount the host dir /pkg/pillar to / in the container, right?

Reply:
it mounts it to the same path in the container, f.e. for me it is: `/home/christoph/projects/eve/pkg/pillar`; then I do:
``` 
cd /path/to/mapped/directory
go test ./...
```

Reply:
Hm... In this case, will the host files in ${EVE_ROOT}/pkg/pillar replaced by the ones in the container?
If that's the case, I'm concerned that IDEs detecting disk changes might go haywire trying to reindex the directory structure that has been modified. Do you have any suggestions on how we could make this process more convenient for the developers?

Reply:
nothing will be replaced, it is just a bind-mount

> Do you have any suggestions on how we could make this process more convenient for the developers?
Use a proper IDE? ;-)

Reply:
It will replace it, but I guess the normal container image has nothing in `/home/user/....`

> I'm not a fan of the "works for me" approach. I would prefer a solution suitable for more people...

Then we should perhaps start first with investigating which IDEs have that problem and where exactly they fail.

P.S.: just feel free to ignore this discussion - you already have the approval from me.

Reply:
Yeah, sure, I hope we can merge the current simple solution.
Let's just use this thread to discuss what could the better solution look like.

So you say you mount `${HOST_EVE_ROOT}/pkg/pillar` into `${HOST_EVE_ROOT}/pkg/pillar` inside the container, right? In this case, you have 2 copies of source code in the container: 1 - /pillar/ - preprocessed code with the patches applied, 2 - `${HOST_EVE_ROOT}`. Right? But in this case, if you change a file on your host - it will be visible only in the second path. So, the patched files are still invisible from the host and the tools/IDEs running there. Right?

Reply:
I normally do it from the eve directory, not pkg/pillar, but the consequences are the same, that I have the code twice in the container.
Yes, it will be only visible in the second path.
Yes, the patched files are invisible from the host and therefore also invisible for the IDEs.

Reply:
> Yeah, sure, I hope we can merge the current simple solution.

Fine from my side.